common-lib/src/main/scala/com/gu/mediaservice/lib/auth/PermissionsHandler.scala

```diff
@@ -0,0 +1,30 @@
+package com.gu.mediaservice.lib.auth
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+object PermissionDeniedError extends Throwable("Permission denied")
+
+trait PermissionsHandler {
+  val failFuture = Future.failed(PermissionDeniedError)
+  val permissionStore: PermissionStore
+
+  def validateUserWithPermissions(user: Principal, permission: PermissionType.PermissionType)
+                                 (implicit ec: ExecutionContext): Future[Principal] =
```

Comment:
Yeah I'm not sure it necessarily makes sense to return a failed future. Why not just a `Future[Boolean]` though?